app/src/main/java/com/example/plantsapp/presentation/ui/notification/broadcastreceiver/NotificationBroadcastReceiver.kt

```diff
@@ -84,6 +85,12 @@ class NotificationBroadcastReceiver : HiltBroadcastReceiver() {
                 putExtra(EXTRA_TASK_FREQUENCY, frequency)
             }
 
+        private var Intent.taskUpdateDate: Long
```

Comment:
You know that Date() is Serialisable? So you can change this extension  to  var Intent.taskUpdateDate: Date, and put serialisable into bundle